src/calculator/index.ts

```diff
@@ -297,48 +300,63 @@ export default class Calculator {
       `${this.chainId}/rounds.json`
     );
 
-    const round = rounds.find((round) => round.id === this.roundId) as Round;
+    const round = rounds.find((round) => round.id === this.roundId);
+
+    if (round === undefined) {
+      throw new ResourceNotFoundError("round");
+    }
 
     const currentResults = await this._calculate(votes);
-    const conversionRates = await Promise.all(
-      potentialVotes.map(async (vote) => [
-        vote.token,
-        await this.priceProvider.getUSDConversionRate(this.chainId, vote.token),
-      ])
-    );
-    const prices = Object.fromEntries(conversionRates) as Record<
-      string,
-      Price & { decimals: number }
-    >;
+    const prices: Record<string, PriceWithDecimals> = {};
```

Comment:
Nit: it can be inferred by the code immediately below, but if you want the reader to save that time, this could be renamed to `usdConversionRateByTokenAddress` or something like that (assuming I'm inferring correctly here).